### DISCUSSION TITLE
How can I index a by etl created multiple document?

### DISCUSSION THREAD

**Discussion by Gomez12** [ravendb/ravendb#15972] [Q&A] [answered]
Currently we are testing some settings with our document modelling.

For speed and convenience we thought we could do this with etl just for testing and to have a complete identical setup between a and b.

For just simple testing it I am following : https://ravendb.net/docs/article-page/5.4/csharp/server/ongoing-tasks/etl/raven#creating-multiple-documents-from-a-single-document
But (to stay within this example) now I don't have any address for an employee anymore. Because my address has got an id of : Employee/100/Address/000000000000001023-B etc.

How can I now load this document for example in an index or for an employee detail page? Because the last part is not predictable for me, and I can't seem to include this in the employee document as well in the ETL process.
Am I doing something wrong?

**Comment by ayende** [comment, marked as answer]
The problem is that you have multiple documents from the same source, and we need to be able to update them.
We don't have the _previous_ state, so we have to delete all of them by prefix and update a new set of documents.
On the other side, this generates a set of unpredictable ids, and the usual way to handle that is to do a `LoadStartingWith` to get the related documents.

So to get the addresses for `employees/100-A` you'll do `session.LoadStartingWith<Address>("employees/100-A/Addresses/")`

**Reply by Gomez12** [reply]
I would think that it is possible to create multiple documents, but then you probably could generate a unique id from a combination of fields or something like that.
But it is not necessary to create multiple documents (for example my use-case is that the current complete document is pretty heavy to generate and feed into raven, while i have a certain subsection which is updated heavy and easy to generate, so we were thinking of generating that as a separate document and then stitch it together with indexes and projections)

I can understand that for generating etc you do a delete with loadstartingwith, but I think it is awkward that you start with an inherent relation between the address and the employee. And when you generate separate documents you lose the ability to fast-load a single document by id.

Especially if the expectation is that you are supposed to generate multiple documents, if you then need to load all the adresses instead of pinpointing one, then what is the use of doing it?
I would expect to get the most out of the function if i can still pinpoint / load separately every created document.

But this are just my thoughts and observations with seeing this functionality.

And for now I will search for a different way to split up the document then.

**Reply by ayende** [reply]
You can't actually do that.
It's perfectly valid to have something like: 

```
"Inventory": [
   {"Name": "Sword" },
   {"Name": "Sword" },
]
```

And you want to ETL them as separate docs. The _values_ are identical, and we have to provide a solution for that.

That said, I don't understand what it is that you are trying to do.
Are you looking for a stable id? For what purpose?
